pkg/server/driver_tidb_test.go

```diff
@@ -95,4 +95,47 @@ func TestConvertColumnInfo(t *testing.T) {
 	}
 	colInfo = column.ConvertColumnInfo(&resultField)
 	require.Equal(t, uint32(4), colInfo.ColumnLength)
+
+	// Test unspecified length
+	for _, tp := range []byte{
+		mysql.TypeBit,
+		mysql.TypeTiny,
+		mysql.TypeShort,
+		mysql.TypeLong,
+		mysql.TypeLonglong,
+		mysql.TypeFloat,
+		mysql.TypeDouble,
+		mysql.TypeNewDecimal,
+		mysql.TypeDuration,
+		mysql.TypeDate,
+		mysql.TypeTimestamp,
+		mysql.TypeDatetime,
+		mysql.TypeYear,
+		mysql.TypeString,
+		mysql.TypeVarchar,
+		mysql.TypeVarString,
+		mysql.TypeTinyBlob,
+		mysql.TypeBlob,
+		mysql.TypeMediumBlob,
+		mysql.TypeLongBlob,
+		mysql.TypeJSON,
+	} {
+		ftb = types.NewFieldTypeBuilder()
+		ftb.SetType(tp)
+		resultField = resolve.ResultField{
+			Column: &model.ColumnInfo{
+				Name:      ast.NewCIStr("a"),
+				ID:        0,
+				Offset:    0,
+				FieldType: ftb.Build(),
+				Comment:   "column a is the first column in table dual",
+			},
+			ColumnAsName: ast.NewCIStr("a"),
+			TableAsName:  ast.NewCIStr("dual"),
+			DBName:       ast.NewCIStr("test"),
+		}
+		colInfo = column.ConvertColumnInfo(&resultField)
+		expectedLen, _ := mysql.GetDefaultFieldLengthAndDecimal(tp)
+		require.NotEqual(t, colInfo.ColumnLength, uint32(expectedLen))
```

Comment:
the PR description says "we can return a non-zero length, at least." so here we can also check `require.NotZero`

BTW why the check is `NotEqual`? I think it should be equal to default field length

Reply:
Nice! I'll fix this.

Reply:
I've modified the code a bit. I did found a bug. Thanks, PTAL.